client/coral-plugin-commentbox/CommentBox.js

```diff
@@ -68,6 +101,15 @@ class CommentBox extends Component {
     this.setState({body: ''});
   }
 
+  addCommentHooks = (hooks = {}) => {
+    if (typeof hooks === 'object') {
```

Comment:
+1 to add the hook to`componentWillUnmount`. 

But I don't see the naming as a confict or issue. Also, I don't see the point of making a list, maybe around performance but it will be tiny. 🤔 

Having a restricted API could lead to issues in the future, that we are not going to experiment, but plugin developers might.

I'll move forward with the list approach, but also keeping in mind that we might need to revisit this again.